packages/serverpod/lib/src/database/adapters/postgres/database_connection.dart

```diff
@@ -542,12 +542,29 @@ class DatabaseConnection {
   }
 
   /// For most cases use the corresponding method in [Database] instead.
-  Future<R> transaction<R>(TransactionFunction<R> transactionFunction) {
+  Future<R> transaction<R>(
+    TransactionFunction<R> transactionFunction, {
+    TransactionSettings? settings,
+  }) {
+    pg.TransactionSettings? transactionSettings;
+    if (settings != null) {
+      transactionSettings = pg.TransactionSettings(
+        isolationLevel: switch (settings.isolationLevel) {
+          IsolationLevel.readCommitted => pg.IsolationLevel.readCommitted,
+          IsolationLevel.readUncommitted => pg.IsolationLevel.readUncommitted,
+          IsolationLevel.repeatableRead => pg.IsolationLevel.repeatableRead,
+          IsolationLevel.serializable => pg.IsolationLevel.serializable,
+          null => null,
+        },
```

Comment:
The TransactionSettings object lives inside of the "Serverpod" domain. It shouldn't have any knowledge of domains using it. Therefore, the adapter should capture the conversion, which is the `database_connection.dart` class. I opted to just do it inline now since there is only a single setting.

But a cleaner solution would maybe be to add a `fromServerpodTransaction` constructor to the driver settings object.

Reply:
Makes sense! Maybe something to do if this method would grow. I think it's fine then to keep as is for now 👍